log_stream.c

```diff
@@ -528,10 +528,21 @@ void mongo_log_stream_response_header(mongo_connection *connection, mongo_cursor
 
 	if (CONTEXT_HAS_NOTIFY_OR_LOG(context, "log_response_header")) {
 		zval **args[3];
-		zval *server, *info;
+		zval *server, *query, *info;
+		int free_query = 0;
 
 		server = php_log_get_server_info(connection);
 
+		/* Command cursors created from a command result document will lack the
+		 * original query document. Allocate a null zval in that case. */
+		if (cursor->query) {
+			query = cursor->query;
+		} else {
+			MAKE_STD_ZVAL(query);
+			ZVAL_NULL(query);
+			free_query = 1;
+		}
```

Comment:
As discussed in chat, we'll leave this code as-is (in the PR) but consider refactoring in https://jira.mongodb.org/browse/PHP-1364